bundles/aliyun-bundle/src/main/java/org/apache/gravitino/oss/credential/OSSTokenProvider.java

```diff
@@ -0,0 +1,268 @@
+          Statement.builder()
+              .effect(Effect.ALLOW)
+              .addAction("oss:PutObject")
+              .addAction("oss:DeleteObject");
+      writeLocations.forEach(
+          location -> {
+            URI uri = URI.create(location);
+            allowPutObjectStatementBuilder.addResource(getOssUriWithArn(arnPrefix, uri));
+          });
+      policyBuilder.addStatement(allowPutObjectStatementBuilder.build());
+    }
+
+    if (!bucketListStatementBuilder.isEmpty()) {
+      bucketListStatementBuilder
+          .values()
+          .forEach(statementBuilder -> policyBuilder.addStatement(statementBuilder.build()));
+    } else {
+      // add list privilege with 0 resources
+      policyBuilder.addStatement(
+          Statement.builder().effect(Effect.ALLOW).addAction("oss:ListBucket").build());
+    }
+    bucketGetLocationStatementBuilder
+        .values()
+        .forEach(statementBuilder -> policyBuilder.addStatement(statementBuilder.build()));
+
+    policyBuilder.addStatement(allowGetObjectStatementBuilder.build());
+    try {
+      return objectMapper.writeValueAsString(policyBuilder.build());
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Condition getCondition(URI uri) {
+    return Condition.builder()
+        .stringLike(
+            StringLike.builder()
+                .addPrefix(concatPathWithSep(trimLeadingSlash(uri.getPath()), "*", "/"))
+                .build())
+        .build();
+  }
+
+  private String getArnPrefix() {
+    if (StringUtils.isNotEmpty(region)) {
+      return "acs:oss:" + region + ":*:";
+    }
+    return "acs:oss:*:*:";
+  }
+
+  private String getBucketName(URI uri) {
+    return uri.getHost();
+  }
+
+  private String getOssUriWithArn(String arnPrefix, URI uri) {
+    return arnPrefix + concatPathWithSep(removeSchemaFromOSSUri(uri), "*", "/");
+  }
+
+  private static String concatPathWithSep(String leftPath, String rightPath, String fileSep) {
+    if (leftPath.endsWith(fileSep) && rightPath.startsWith(fileSep)) {
+      return leftPath + rightPath.substring(1);
+    } else if (!leftPath.endsWith(fileSep) && !rightPath.startsWith(fileSep)) {
+      return leftPath + fileSep + rightPath;
+    } else {
+      return leftPath + rightPath;
+    }
+  }
+
+  // Transform 'oss://bucket/path' to /bucket/path
+  private String removeSchemaFromOSSUri(URI uri) {
+    String bucket = uri.getHost();
+    String path = trimLeadingSlash(uri.getPath());
+    return String.join(
+        "/", Stream.of(bucket, path).filter(Objects::nonNull).toArray(String[]::new));
+  }
+
+  private String trimLeadingSlash(String path) {
+    return path.startsWith("/") ? path.substring(1) : path;
+  }
+
+  private String getRoleName(String userName) {
+    return "gravitino_" + userName;
+  }
+
+  /**
+   * Closes this stream and releases any system resources associated with it. If the stream is
```

Comment:
It's a little odd to add these comments since `close` did nothing here

Reply:
From parent class, automatically added